addons/mrp/wizard/mrp_copy_bom_operations.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models, fields, _
+
+
+class CopyBOMOperations(models.TransientModel):
+    _name = 'mrp.copy.bom.operations'
+    _description = 'Copy Bill of Material Operations'
+
+    bom_id = fields.Many2one('mrp.bom')
+    operation_ids = fields.Many2many('mrp.routing.workcenter')
+
+    @api.model
+    def default_get(self, field_names):
+        res = super().default_get(field_names)
+        active_id = self.env.context.get('active_id')
+        res['bom_id'] = active_id
+        return res
+
+    def action_copy_operations(self):
+        commands = []
+        for operation in self.operation_ids:
+            duplicated_operation = operation.copy({'name': _("%s (copy)", operation.name), 'bom_id': self.bom_id.id})
+            commands.append((4, duplicated_operation.id))
+        self.bom_id.update({'operation_ids': commands})
```

Comment:
As `operation_ids` is the O2M counterpart of the operation's M2O field `bom_id`, I don't think you need to set it on both BOM and operations.
The operation copy with the default BOM id should be enough.